orix/plot/unit_cell_plot.py

```diff
@@ -0,0 +1,199 @@
+    """Calculate cell vertices for unit cells."""
+    vectors = np.asarray(vectors)
+    assert vectors.shape == (3, 3), "Vectors must be (3, 3) array."
+    # generate list of lattice basis vectors from (000) to (111) (positive only)
+    verts = np.array(list(product(*zip((0, 0, 0), (1, 1, 1)))))
+    verts = (verts[..., np.newaxis] * vectors).sum(axis=1)
+    center = verts.mean(axis=0)
+    return verts - center  # center on (0, 0, 0)
+
+
+def _calculate_basic_unit_cell_edges(verts, vectors):
+    """Calculate valid unit cell edges for unit cells."""
+    vectors = np.asarray(vectors)
+    assert vectors.shape == (3, 3), "Vectors must be (3, 3) array."
+    a1, a2, a3 = np.linalg.norm(vectors, axis=-1)
+    # get valid edges from all unit cell egde possibilities unit cell
+    edges_valid = []
+    # for all possible combinations of vertices, keep if the distance between them is
+    # equal to any of the basis vectors
+    for v1, v2 in combinations(verts, 2):
+        if np.isclose((a1, a2, a3), np.linalg.norm(v2 - v1)).any():
+            # extra case for hexagonal unit cells, do not plot (0000)-(11-20) edge
+            # in hexagonal cells the gamma angle is 120 degrees (between a1 and a2)
+            # if xy of (v2 - v1) is equal to xy of vectors[0] + vectors[1] -> skip
+            if np.isclose((v2 - v1)[:-1], (vectors[0] + vectors[1])[:-1]).all():
+                continue
+            edges_valid.append((v1, v2))
+    return np.array(edges_valid)
+
+
+def _plot_unit_cell(
+    rotation,
+    c="tab:blue",
+    axes_length=0.5,
+    structure=None,
+    crystal_reference_frame_axes_position="origin",
+    **arrow_kwargs,
+):
+    """Plot the unit cell orientation, showing the sample and crystal reference frames.
+
+    Parameters
+    ----------
+    rotation : orix.quaternion.Rotation
+        Rotation of the unit cell.
+    c : str, optional
+        Unit cell edge color.
+    axes_length : float, optional
+        Length of the reference axes in Angstroms, by default 0.5.
+    structure : diffpy.structure.Structure or None, optional
+        Structure of the unit cell, only orthorhombic lattices are currently
+        supported. If not given, a cubic unit cell with a lattice parameter of
+        2 Angstroms will be plotted.
+    crystal_reference_frame_axes_position : str, optional
+        Plot the crystal reference frame axes at the "origin" or "center" of the
+        plotted cell.
+    arrow_kwargs : dict, optional
+        Keyword arguments passed to
+        :class:`matplotlib.patches.FancyArrowPatch`, for example "arrowstyle".
+
+    Returns
+    -------
+    fig : matplotlib.figure.Figure
+        The plotted figure.
+    """
+    # requires active rotation of the lattice in the sample reference frame
+    rotation = ~rotation
+
+    crystal_reference_frame_axes_position = (
+        crystal_reference_frame_axes_position.lower()
+    )
+    if not crystal_reference_frame_axes_position in ("origin", "center"):
```

Comment:
PyCharm informs me that this should be

```suggestion
    if crystal_reference_frame_axes_position not in ("origin", "center"):
```